docs-src/java/how-to-use-worker-versioning-in-java.md

```diff
@@ -0,0 +1,106 @@
+---
+id: how-to-use-worker-versioning-in-java
+title: How to use Worker Versioning in Java
+sidebar_label: Worker Versioning
+description: Learn how to version your Java Workers by using build-ID based versioning
+tags:
+  - java
+  - how-to
+---
+
+To make use of Worker Versioning in Java, you will need to do the following:
+
+1. Determine and assign a Build ID to your built Worker code, and opt in to versioning
+2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID
+
+### Assign a Build ID to your Worker
+
+Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a
+reasonable choice as Build ID). To assign it in your Worker code, you'd assign
+the following Worker Options:
+
+```java
+// ...
+WorkerOptions workerOptions = WorkerOptions.newBuilder()
+    .setBuildId("deadbeef")
+    .setUseBuildIdForVersioning(true)
+    // ...
+    .build();
+Worker w = workerFactory.newWorker("your_task_queue_name", workerOptions);
+// ...
+```
+
+That's all you need to do in your Worker code. Importantly, if you start this Worker, it won't
+receive any tasks. That's because you need to tell the Task Queue about your Worker's Build ID
+first.
```

Comment:
```suggestion
That's all you need to do in your Worker code.
Importantly, if you start this Worker, it won't receive any tasks.
That's because you need to tell the Task Queue about your Worker's Build ID first.
```